File: primerTrimestre/introduccion_programacion_python/2alternativas_py/ej6alt.py
# Programa: ej6alt.py
# Propósito: Programa que lea una cadena por teclado y compruebe si es una letra mayúscula.
# Autor: Antonio Quesada
# Fecha: 23/10/2019.
#
# Variables a usar:
#   * cadena
#
# Algoritmo:
# La cadena debe tener una longitud de uno para introducir una letra. 
# Compara si esta entre A y Z mayúscula y si no es el caso la letra es minúscula.

# Petición de datos
cadena = str(input("Introduce una cadena: "))
print("------------------------------------------")


# Proceso y Salida
if len(cadena) == 1 and cadena >= 'A' and cadena <= 'Z'or cadena == 'Á' or cadena == 'É' or cadena == 'Í' or cadena == 'Ó' or cadena == 'Ú': 
    print("Se trata de una letra mayúscula")
else: 
    print("No es una letra mayúscula, es minúscula")

File: primerTrimestre/java basico/2alternativas/Ej6alt.java
/**Programa que lea una cadena por teclado y compruebe si es una letra mayúscula.
 * 
 * @author antonio
 */

import java.util.Scanner;

public class Ej6alt {
  public static void main(String[] args) {
    Scanner s = new Scanner(System.in);
    
    //Definimos cadena.
    String cadena;

    // Petición de datos
    System.out.print("Introduce una cadena: ");
    cadena = s.nextLine();
    
    //La cadena debe tener una longitud de una letra (función length) y de comparar que este entre A y Z mayúscula.
    if (cadena.length()==1 && (cadena.compareTo("A")>=0 && cadena.compareTo("Z")<=0)) {
      System.out.println("La cadena es una letra mayúscula");
      
      //En el caso de que no sea mayúscula mostrará que es minúscula. 
    } else {
      System.out.println("La cadena es una letra minúscula");
    }
    
    
  }

}
